h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -751,17 +802,46 @@ public G(int ncols, AGG[] aggs, boolean hasMedian) {
         }
       }
     }
+    
+    private int colsCount() { return _gs.length + _gsStr.length; }
 
     public G fill(int row, Chunk chks[]) {
-      for (int c = 0; c < chks.length; c++) // For all selection cols
-        _gs[c] = chks[c].atd(row); // Load into working array
+      for (int c = 0; c < chks.length; c++) { // For all selection cols
+        Vec vec = chks[c].vec();
+
+        // Load into working array
+        if (vec.isString())
+          _gsStr[c] = chks[c].atStr(new BufferedString(), row);
+        else
+          _gs[c] = chks[c].atd(row);
+      }
+      
       _hash = hash();
       return this;
     }
 
-    public G fill(int row, Chunk chks[], int cols[]) {
+    public G fill(int row, Chunk[] chks, int[] colsNum) {
+      int[] gbColsTypes = new int[colsNum.length];
+      Arrays.fill(gbColsTypes, Vec.T_NUM);
+
+      return fill(row, chks, colsNum, gbColsTypes);
+    }
+
+    public G fill(int row, Chunk[] chks, int[] cols, int[] gbColsTypes) {
+      int strIdx = 0;
+      int numIdx = 0;
+      
       for (int c = 0; c < cols.length; c++) // For all selection cols
-        _gs[c] = chks[cols[c]].atd(row); // Load into working array
+        // Load into working array
+        try {
+          if (gbColsTypes[c] == Vec.T_STR)
+            _gsStr[strIdx++] = chks[cols[c]].atStr(new BufferedString(), row);
+          else
+            _gs[numIdx++] = chks[cols[c]].atd(row);
+        }
+      catch (Exception e) {
```

Comment:
why are you catching exception here?

Reply:
Sorry, I missed this. It is a remnant from debugging. I removed it.